pkg/dynatrace/dynatrace.go

```diff
@@ -151,6 +151,31 @@ func (dynatraceApiClient *DynatraceApiClient) MakeRequest(method, path string, r
 	return dynatraceApiClient.httpClient.Do(req)
 }
 
+func (dynatraceApiClient *DynatraceApiClient) GetListDynatraceHttpMonitors(clusterId string) (ExistsHttpMonitorInDynatraceResponse, error) {
+	var existsHttpMonitorResponse ExistsHttpMonitorInDynatraceResponse
+
+	path := fmt.Sprintf("/synthetic/monitors/?tag=cluster-id:%s", clusterId)
+	resp, err := dynatraceApiClient.MakeRequest(http.MethodGet, path, "")
+	if err != nil {
+		return existsHttpMonitorResponse, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return existsHttpMonitorResponse, fmt.Errorf("failed to fetch monitor in Dynatrace. Status code: %d", resp.StatusCode)
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&existsHttpMonitorResponse); err != nil {
+		return existsHttpMonitorResponse, fmt.Errorf("error parsing JSON: %w", err)
+	}
+
+	if len(existsHttpMonitorResponse.Monitors) == 0 {
+		return existsHttpMonitorResponse, nil
+	}
+
+	return existsHttpMonitorResponse, nil
```

Comment:
Aren't these the same things? You marshall on line 168 the json into the type, then on 172 return that object if the length is 0, and then on line 176 you also return that same object if > 0

Reply:
This has been fixed